enterprise/course_catalog_api.py

```diff
@@ -158,20 +160,23 @@ def get_common_course_modes(self, course_run_ids):
 
         return available_course_modes
 
-    def _load_data(self, resource, **kwargs):
+    def _load_data(self, resource, default=DEFAULT_VALUE_SAFEGUARD, **kwargs):
         """
         Load data from API client.
 
         Arguments:
-            resource(str): type of resource to load
+            resource(string): type of resource to load
+            default(any): value to return if API query returned empty result. Sensible values: [], {}, None etc.
```

Comment:
While this works fine, I wonder whether we shouldn't change `get_edx_api_data` from edx/edx-platform instead. It currently [always returns an empty list when it cannot contact the API](https://github.com/edx/edx-platform/blob/424b4356a583fccfc9f5dd64b387e39911aecc5c/openedx/core/lib/edx_api_utils.py#L78-L84), which makes sense when requesting collections, but makes no sense when requesting a particular resource (by providing the `resource_id` parameter).